proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -515,4 +522,54 @@ private boolean isIncognitoEnabled(@Nonnull final Map<String, Object> inputConte
                 .map(value -> value.equalsIgnoreCase("true"))
                 .orElse(false);
     }
+
+    private void populateProperties(
+            final String testName,
+            final TestBucket testBucket,
+            final Map<String, PayloadProperty> testProperties) {
+        final Payload payload = testBucket.getPayload();
+        if (payload != null && payload.getJson() != null) {
+            payload.getJson()
+                    .fields()
+                    .forEachRemaining(
+                            field -> {
```

Comment:
nit: this seems worth a dedicated function

Reply:
Moved to separate function